sources/packages/backend/apps/api/src/route-controllers/education-program/_tests_/e2e/education-program.institutions.controller.createEducationProgram.e2e-spec.ts

```diff
@@ -0,0 +1,173 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import {
+  Institution,
+  InstitutionLocation,
+  ProgramStatus,
+  User,
+} from "@sims/sims-db";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeInstitutionLocation,
+  getProviderInstanceForModule,
+} from "@sims/test-utils";
+import {
+  createTestingAppModule,
+  BEARER_AUTH_TYPE,
+  InstitutionTokenTypes,
+  getInstitutionToken,
+  authorizeUserTokenForLocation,
+  getAuthRelatedEntities,
+} from "../../../../testHelpers";
+import * as request from "supertest";
+import { FormService } from "../../../../services";
+import { DryRunSubmissionResult } from "../../../../types";
+import { TestingModule } from "@nestjs/testing";
+import { AppInstitutionsModule } from "../../../../app.institutions.module";
+
+describe("EducationProgramOfferingInstitutionsController(e2e)-createOffering", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let collegeF: Institution;
+  let collegeFLocation: InstitutionLocation;
+  let collegeFUser: User;
+  let testingModule: TestingModule;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource, module } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    const { institution, user: institutionUser } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeF = institution;
+    collegeFLocation = createFakeInstitutionLocation(collegeF);
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+    collegeFUser = institutionUser;
+    testingModule = module;
+  });
+
+  it("Should create an education program when valid data is passed.", async () => {
+    // Arrange
+    const payload = {
+      name: "Education Program test",
+      description: "Education Program description...",
+      credentialType: "undergraduateCertificate",
+      cipCode: "11.1111",
+      fieldOfStudyCode: "15",
+      nocCode: "2174",
+      sabcCode: "PQP0",
+      institutionProgramCode: "DDD",
+      programIntensity: "Full Time and Part Time",
+      programDeliveryTypes: {
+        deliveredOnSite: true,
+        deliveredOnline: false,
+      },
+      completionYears: "12WeeksToLessThan1Year",
+      courseLoadCalculation: "credit",
+      regulatoryBody: "ptib",
+      entranceRequirements: {
+        minHighSchool: true,
+        hasMinimumAge: true,
+        requirementsByInstitution: true,
+        requirementsByBCITA: true,
+      },
+      eslEligibility: "lessThan20",
+      hasJointInstitution: "no",
+      hasWILComponent: "no",
+      hasTravel: "no",
+      hasIntlExchange: "no",
+      programDeclaration: true,
+      totalOfferings: "",
+      isBCPrivate: "",
```

Comment:
Same as https://github.com/bcgov/SIMS/pull/2012/files#r1231552660 for all empty variables